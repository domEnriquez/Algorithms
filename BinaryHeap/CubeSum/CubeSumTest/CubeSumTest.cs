using ApprovalTests;
using ApprovalTests.Reporters;
using CubeSum;
using NUnit.Framework;
using System;
using System.IO;
using System.Text;

namespace CubeSumTest
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class CubeSumTest
    {
        CubeSumBinaryHeap cs;
        StringBuilder fakeOutput;

        [SetUp]
        public void SetUp()
        {
            cs = new CubeSumBinaryHeap();
            fakeOutput = new StringBuilder();
            Console.SetOut(new StringWriter(fakeOutput));
        }

        [Test]
        public void showSortedCubeSumOf0()
        {
            cs.ShowCubeSums(0);

            Approvals.Verify(fakeOutput);
        }

        [Test]
        public void showSortedCubeSumOf0To2()
        {
            cs.ShowCubeSums(2);

            Approvals.Verify(fakeOutput);
        }

        [Test]
        public void showSortedCubeSumof0To5()
        {
            cs.ShowCubeSums(5);

            Approvals.Verify(fakeOutput);
        }

        [Test]
        public void showSortedCubeSumof0to12()
        {
            cs.ShowCubeSums(12);

            Approvals.Verify(fakeOutput);
        }

        //All distinct integers a, b, c, and d between 0 and 10^6 such that a^3 + b^3 = c^3 + d^3
        [Test]
        public void showCubeSumsObtainedInTwoWays() 
        {
            cs.CubeSumsObtainedInTwoWays(12);

            Approvals.Verify(fakeOutput);
        }

        [Test]
        public void showCubeSumsObtainedInTwoWaysOf0To1000()
        {
            cs.CubeSumsObtainedInTwoWays(1000);

            Approvals.Verify(fakeOutput);
        }
    }
}
