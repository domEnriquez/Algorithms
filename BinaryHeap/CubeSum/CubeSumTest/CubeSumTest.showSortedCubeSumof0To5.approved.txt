0 = 0^3 + 0^3
1 = 1^3 + 0^3
1 = 0^3 + 1^3
2 = 1^3 + 1^3
8 = 2^3 + 0^3
8 = 0^3 + 2^3
9 = 1^3 + 2^3
9 = 2^3 + 1^3
16 = 2^3 + 2^3
27 = 3^3 + 0^3
27 = 0^3 + 3^3
28 = 1^3 + 3^3
28 = 3^3 + 1^3
35 = 2^3 + 3^3
35 = 3^3 + 2^3
54 = 3^3 + 3^3
64 = 4^3 + 0^3
64 = 0^3 + 4^3
65 = 1^3 + 4^3
65 = 4^3 + 1^3
72 = 2^3 + 4^3
72 = 4^3 + 2^3
91 = 3^3 + 4^3
91 = 4^3 + 3^3
125 = 5^3 + 0^3
125 = 0^3 + 5^3
126 = 1^3 + 5^3
126 = 5^3 + 1^3
128 = 4^3 + 4^3
133 = 2^3 + 5^3
133 = 5^3 + 2^3
152 = 3^3 + 5^3
152 = 5^3 + 3^3
189 = 4^3 + 5^3
189 = 5^3 + 4^3
250 = 5^3 + 5^3
